src/main/java/org/mockito/ArgumentCaptor.java

```diff
@@ -100,12 +84,41 @@ private ArgumentCaptor(Class<? extends T> clazz) {
      * @return null or default values
      */
     public T capture() {
-        Mockito.argThat(capturingMatcher);
+        return captureIf(Any.ANY);
```

Comment:
In the description of  #358 I wrote about the unexpected TooManyActualInvocations-Exception that lead me to open the ticket. We can get rid of it in case the captor is created via `ArgumentCaptor.forClass(Class)` cause we have the class-Object. Something like this shoud do the trick: 
```
if (clazz==null)
   return captureIf( Any.ANY);
return captureIf(isNullOrA(clazz));
```

This would avoid the capturing of arguments that are not an instances of the given type. Obviously we can't  use `isA(Class)` here because _null_ arguments must be captured too. 

WDYT?